ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ProcessingLogConfig.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.util;
+
+import io.confluent.ksql.util.KsqlConfig;
+import java.util.Map;
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigDef.ValidString;
+import org.apache.kafka.common.config.ConfigDef.Validator;
+
+public class ProcessingLogConfig extends AbstractConfig {
+  private static final String PROPERTY_PREFIX = "processing.log.";
+
+  private static String propertyName(final String name) {
+    return KsqlConfig.KSQL_CONFIG_PROPERTY_PREFIX + PROPERTY_PREFIX + name;
+  }
+
+  public static final String AUTO_CREATE_ON = "on";
+  public static final String AUTO_CREATE_OFF = "off";
+
+  public static final String STREAM_NAME = propertyName("stream.name");
+  private static final String STREAM_NAME_DEFAULT = "PROCESSING_LOG";
+  private static final String STREAM_NAME_DOC =
+      "If automatic processing log stream creation is enabled, KSQL sets the name of the "
+          + "stream to the value of this property.";
+
+  public static final String TOPIC_NAME = propertyName("topic.name");
+  private static final String TOPIC_NAME_DEFAULT = "ksql_processing_log";
```

Comment:
I wouldn't consider this an internal topic - its reasonable for a user to want to consume it directly. Originally I had it prepend cluster ID and reversed that since it made more sense to me to just use whatever topic name the user had configured. However to avoid conflicts I think it would make sense to prepend the cluster ID if the user hasn't explicitly configured a name. I'll fix that.